fhir-operation-bulkdata/src/main/java/com/ibm/fhir/operation/bulkdata/client/BulkDataClient.java

```diff
@@ -312,11 +309,139 @@ public PollingLocationResponse status(String job) throws Exception {
 
     /**
      * @param job
-     * @return
+     * @return status code
      * @throws Exception
      */
-    public void delete(String job) throws Exception {
-        // Example: https://localhost:9443/ibm/api/batch/jobexecutions/9
+    public Response.Status delete(String job) throws Exception {
+        // <b>Code Flow</b>
+        // 1 - Check if the Job Exists and is Valid
+        // 2 - Check if the Tenant is expected
+        // 3 - Try to immediately DELETE (Optimistic)
+        //      A - SC_NO_CONTENT = SUCCESS GO_TO End
+        //      B - !SC_INTERNAL_ERROR GO_TO STOP 
+        //      C - ERROR GO_TO Error 
+        // 4 - Try to stop the PUT /ibm/api/batch/jobinstances/<job>?action=stop
+        //     A - SC_CONTENT_REDIRECT (302) GO_TO Location to PUT
+        //       - Stop requests sent to the batch REST API must be sent directly to the executor where the job is running.
+        //       - Link - https://www.ibm.com/support/knowledgecenter/SSEQTP_liberty/com.ibm.websphere.wlp.doc/ae/rwlp_batch_rest_api.html#rwlp_batch_rest_api__STOP_requests
+        //     B - SC_ACCEPTED (202) GO_TO ACCEPTED
+        // RETURN - ACCEPTED: SC_ACCEPTED - STILL PROCESSING
+        // RETURN - DONE: SC_OK - DONE DELETING
```

Comment:
maybe we should propose change to the bulkdata spec to add the 200 OK response for deletion. currently, in the spec, there is only 202 accepted for deletion.

Reply:
I completely agree.  Lee said they is some discussion in the bulkdata workgroup on this behavior.